.repo/sample-list-generator/src/libs/sample-collector.ts

```diff
@@ -0,0 +1,72 @@
+import path from 'path';
+import fs from 'fs/promises';
+import { AVAILABLE_FOLDERS, REPO_BASE_URL } from '../constants';
+import { getApiListForSample } from './api-detector';
+import type { AvailableFolderItem, SampleItem } from '../types';
+import { getBasePath, isFileExists } from '../utils/filesystem';
+import { getManifest } from '../utils/manifest';
+
+export const getAllSamples = async () => {
+  let samples: SampleItem[] = [];
+
+  // loop through all available folders
+  // e.g. api-samples, functional-samples
+  for (let samplesFolder of AVAILABLE_FOLDERS) {
+    const currentSamples = await getSamples(
+      samplesFolder.path,
+      samplesFolder.type
+    );
+    samples.push(...currentSamples);
+  }
+
+  return samples;
+};
+
+const getSamples = async (
+  currentRootFolderPath: string,
+  sampleType: AvailableFolderItem['type']
+): Promise<SampleItem[]> => {
+  const samples: SampleItem[] = [];
+  const basePath = getBasePath();
+
+  // get all contents in the folder
+  const contents = await fs.readdir(path.join(basePath, currentRootFolderPath));
+
+  for (let content of contents) {
+    const currentPath = path.join(basePath, currentRootFolderPath, content);
+
+    // if content is not a folder, skip
+    if (!(await fs.stat(currentPath)).isDirectory()) {
```

Comment:
maybe turn this into a helper in filesystem?

Reply:
Done. https://github.com/GoogleChrome/chrome-extensions-samples/pull/963/commits/2bd89a80506df4747af88a394f9f7e15f212bf76